jina/flow/base.py

```diff
@@ -424,7 +424,21 @@ def __call__(self, *args, **kwargs):
         return self.build(*args, **kwargs)
 
     def __enter__(self):
-        return self.start()
+        class CatchAllCleanupContextManager():
```

Comment:
a try catch exit should be enough right?

Reply:
No it is not. The point of the context manager is, that it will execute the cleanup code, even thought the main code is interrupted